addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -637,6 +637,13 @@ options.Class.include({
         this.rpc = this.bindService("rpc");
         this.orm = this.bindService("orm");
     },
+    /**
+     * @override
+     */
+    destroy() {
+        options.clearRpcCache();
+        this._super(...arguments);
+    },
```

Comment:
- As requested, you define the function in web_editor instead of website, but you should also use it over there. The web_editor implementation should define the function + call the clearing of the cache at the right time.
- You  don't want to clear the cache when destroying each option. There are many of them and they are destroyed during edition. You should do as the m2o cache: only once when leaving edit mode? Unless I am missing something. Also why no sharing the m2o cache? Can't you merge both? :)